docs/contribution/patterns.md

```diff
@@ -0,0 +1,3 @@
+# Patterns contributions
+
+Patterns contribution guidelines coming soon!
```

Comment:
The IBM.com Library team has a ton of patterns work on our roadmap very soon, as this is a huge need for IBM.com adopters (React/Vanilla). Our current plan is creating a patterns package in our monorepo. We should sync up to see how much this will align with what Carbon team has in mind.
cc: @photodow @oliviaflory @larahanlon2

Reply:
WH also has a ton of Pattern work going on as well @AT-public @vincentsnagg @janedepgen